torchao/csrc/cuda/tensor_core_tiled_layout/tensor_core_tiled_layout.cu

```diff
@@ -123,11 +164,18 @@ __global__ void _dequantize_int4_kernel(
       // All b values within a 16x16 tile should fall within the same q group
       // Hence we load 1 scale and zero per loop
       int qgroup = ks[0] /  groupSize;
-      const __nv_bfloat16 *pSZ = reinterpret_cast<const __nv_bfloat16*>(&scales_and_zeros.value()[qgroup][n0][0]);
-
-      // Vectorize scales and zeros
-      __nv_bfloat162 scale2 = __bfloat162bfloat162(pSZ[0]);
-      __nv_bfloat162 zero2 = __bfloat162bfloat162(pSZ[1]);
+      if (scales_and_zeros) {
+        const auto&sz = *scales_and_zeros;
```

Comment:
@lcskrishna This change fixes compilation for me, please refer to my scratch space : https://github.com/petrex/ao/tree/rocm_tensor_tile

Reply:
Thanks @petrex for taking a look. I will update it accordingly.